test/e2e/revision_test.go

```diff
@@ -76,18 +93,34 @@ func (test *e2eTest) revisionListWithService(t *testing.T, serviceNames ...strin
 	}
 }
 
-func (test *e2eTest) revisionDelete(t *testing.T, serviceName string) {
-	revName := test.findRevision(t, serviceName)
-
+func (test *e2eTest) revisionDelete(t *testing.T, revName string) {
 	out, err := test.kn.RunWithOpts([]string{"revision", "delete", revName}, runOpts{})
 	assert.NilError(t, err)
 
 	assert.Check(t, util.ContainsAll(out, "Revision", revName, "deleted", "namespace", test.kn.namespace))
 }
 
-func (test *e2eTest) revisionDescribeWithPrintFlags(t *testing.T, serviceName string) {
-	revName := test.findRevision(t, serviceName)
+func (test *e2eTest) revisionMultipleDelete(t *testing.T, revisionNames []string) {
+	existRevision1 := revisionNames[0]
+	existRevision2 := revisionNames[1]
+	nonexistRevision := revisionNames[2]
+	out, err := test.kn.RunWithOpts([]string{"revision", "list"}, runOpts{NoNamespace: false})
+	assert.NilError(t, err)
+	assert.Check(t, strings.Contains(out, existRevision1), "Required revision1 does not exist")
+	assert.Check(t, strings.Contains(out, existRevision2), "Required revision2 does not exist")
+	assert.Check(t, !strings.Contains(out, nonexistRevision), "Nonexistent revision does exist")
```

Comment:
I think this block is not required as if the revision doesnt exist, it should be tested in delete check below.

Reply:
I'd rather keep it as if the tests before this one are moved around there might not be 2 revisions. Though I'll remove it if you feel strongly.

Reply:
In that case, your tests should generate revisions by its own and not rely on generating test artifacts by other tests.